pkg/nfd-worker/nfd-worker.go

```diff
@@ -312,6 +299,33 @@ func (w *nfdWorker) Run() error {
 
 	w.ownerReference = ownerReference
 
+	return nil
+}
+
+// Run NfdWorker client. Returns an error if a fatal error is encountered, or, after
+// one request if OneShot is set to 'true' in the worker args.
+func (w *nfdWorker) Run() error {
+	klog.InfoS("Node Feature Discovery Worker", "version", version.Get(), "nodeName", utils.NodeName(), "namespace", w.kubernetesNamespace)
+
+	// Read configuration file
+	err := w.configure(w.configFilePath, w.args.Options)
+	if err != nil {
+		return err
+	}
+
+	// Create ticker for feature discovery and run feature discovery once before the loop.
+	labelTrigger := infiniteTicker{Ticker: time.NewTicker(1)}
+	labelTrigger.Reset(w.config.Core.SleepInterval.Duration)
+	defer labelTrigger.Stop()
+
+	// Create owner ref
+	if !w.config.Core.NoOwnerRefs {
```

Comment:
Hmm, now that taking a second look, I'd suggest to move this to `configureCore` instead. This also better serves the future in case we re-add dynamic reconfigurability (say via a CRD).